### PR TITLE
React construct

### PR DESCRIPTION
**Pull request by mnapoli**
This PR adds a `react` website construct.

This construct is heavily based on the `static-website` construct.

Why a separate React construct? Yes, it is possible to deploy a React application with the existing [static website construct](https://github.com/getlift/lift/blob/master/docs/static-website.md). That new construct is preconfigured specifically for React. It is slightly more practical.

The difference will not especially be in technical terms. It is an attempt at being even closer to the real use cases of developers. In terms of adoption, we also want to see if we can get more attention from the React community by adapting serverless to _their_ world, instead of adapting React to the serverless world.

**In other words, starting from users and communities instead of starting from the technical solution.**

Example:

```yaml
service: my-app
provider:
    name: aws

constructs:
    website:
        type: react-website

plugins:
    - serverless-lift
```

On `serverless deploy`, the application will be built (via `npm run build`) and deployed as a public website.

[Full documentation](https://github.com/getlift/lift/blob/react/docs/react-website.md)

---

On a side note, this was an interesting first step into construct reuse (which could be big with #23).

### CODE REVIEW

**Comment by piolet** [comment]
It also works for VueJS ou Angular ?
it is not specific to React a priori?
Maybe rename the builder to a more general purpose one?

**Comment by mnapoli** [comment]
@piolet we already have `static-website` for that. It is specific to React.

**Comment by piolet** [comment]
@mnapoli I don't understand your answer.
If there is already static-website for VueJS or Angular, why make a specific one for React ?

The difference is that the `npm run build` command is launched before the upload on S3 ? Well, this command also works for VueJS or Angular if I'm not wrong.

So why make a specific construct for React when it can also be applied to another framework?

**Comment by mnapoli** [comment]
@piolet I answered that in the pull request description, is it still confusing to you?

**Comment by piolet** [comment]
my remark is right: why name this construct "react-website" when without any other modification, it could also apply a VueJS or Angular application.

Or I didn't understand the description actually.

Tell me if I'm wrong, but the idea of this construct is to launch the build command, then get the dist directory to send it to S3, right?

**Comment by mnapoli** [comment]
For VueJS the path of files to upload will be different. Also the dev commands might be different.

But also, from the description:

> The difference will not especially be in technical terms. It is an attempt at being even closer to the real use cases of developers. In terms of adoption, we also want to see if we can get more attention from the React community by adapting serverless to their world, instead of adapting React to the serverless world.
>
> In other words, starting from users and communities instead of starting from the technical solution.

**Comment by mnapoli** [comment]
> I think the DX is nice but it kind of feels like trying to solve with code what may be solved with documentation.

That is a very very good point… I'll ponder on this. So far I haven't seen much enthusiasm about that approach, both internally and externally, so I'm pausing on this and thinking a bit more. Your idea may be a good approach 🤔

**Comment by mnapoli** [comment]
Closing for now, I don't think this makes sense anymore.
